### PR TITLE
[vendor-change] Add functions for prototyping Vault signing CSR

### PR DESCRIPTION
**Pull request by lei-tang**
Add functions for prototyping Vault signing CSR, includes:
- Create a connection to Vault
- Mount Vault PKI
- Set CA signing key and cert
- Set workload role for issuing certificates
- Sign CSR
- Run a prototype that excercises the flow from creating a connection 
to Vault to signing a CSR and printing the certificate generated.

### CODE REVIEW

File: security/pkg/adapter/vault/ca.go [hunk 1]
```diff
@@ -17,11 +17,31 @@ package vault
...
+	configCaKeyCertPath = "istio_ca/config/ca"
+	workloadRolePath    = "istio_ca/roles/workload_role"
+	signCsrPath         = "istio_ca/sign-verbatim"
+)
+
+// Config for prototyping purpose
+const (
+	vaultAddrForTesting = "http://127.0.0.1:8200"
+	tokenForTesting     = "myroot"
+	testCAKeyCertFile   = "testdata/istio_ca.pem"
```
Comment (myidpt) [inline security/pkg/adapter/vault/ca.go]: It's OK to put it here for a prototype. This can be moved to ca_test.go in the following PRs.

**Comment by myidpt** [comment]
Please fix the lint errors. Ping me back when all required tests pass, I will add /lgtm :)

**Comment by lei-tang** [comment]
@myidpt Thanks! The tests failed because the dependencies related to Hashi Vault have not been checked into the repo yet. I have created a PR https://github.com/istio/vendor-istio/pull/55 to add the dependencies related to Hashicorp Vault.

**Comment by kyessenov** [comment]
@lei-tang To use the new vendor, you should:
1. push your PR to an upstream branch in istio/vendor-istio, so that SHA is visible
2. use that SHA here temporarily, validate that tests pass
3. once we merge vendor PR, you need to update vendor SHA here to the new merged one
4. we merge this PR.

File: security/pkg/adapter/vault/ca.go (hunk 1 above)
Comment (incfly) [reply security/pkg/adapter/vault/ca.go]: +1 to move to xx_test.go, could you leave a TODO?

**Comment by lei-tang** [comment]
The vendor SHA in this PR has been updated to the SHA of the newly merged dependency (https://github.com/istio/vendor-istio/pull/55).

**Comment by ldemailly** [comment]
remember the vendor PR needs to be merged and this one updated before it can be merged

**Comment by ldemailly** [comment]
vendor merged - thx!
let's update this and merge it

**Comment by lei-tang** [comment]
This PR has been updated to use the vendor in the merged vendor dependency (https://github.com/istio/vendor-istio/commit/1877ab0272b14e8cc52c9ba774edad85ee891941).

**Comment by ldemailly** [comment]
closing in favor of #4984

**Pull request closed without merging**